src/classes/CRLP_FiscalYears.cls

```diff
@@ -63,6 +63,16 @@ public class CRLP_FiscalYears {
         }
     }
 
+    /*********************************************************************************************************
+    * @description Returns the current year, fiscal or calendar depending on the rollup setting as a formatted string.
+    All punctuation will have been removed
+    * @param targetDate The date for which to look up the year.
+    * @return String The calendar or fiscal year for the given Date as a string.
+    */
+    public String getYearFormattedString(Date targetDate) {
```

Comment:
Since this class already has a static method for `getFiscalYear(Date targetDate)`, you could create a new static method for `getYearAsFormattedString(Date theDate)`. This way all the places that call `UTIL_String.removeNonNumeric(closeDate.addMonths(n).year().format())` could just be `CRLP_FiscalYears.getYearAsFormattedString(closeDate)`. The instance method can be kept (I believe that's allowed), though it should probably be renamed to `getYearAsFormattedString`.

Reply:
Do you think it's worth making this change at this time?  It's slightly more invasive and requires a bit more code rework, and I'd need to be I understand the business rules around the dates being passed into the function.  Just want to be sure I don't break something.

Reply:
The `year().format()` logic is acting upon other dates, not just the current fiscal year.  Right now we don't really have a use for new static method. Even the current static method is only used within the `CRLP_FiscalYears` class.  I can see how a method that accepts a Date and return the year as a string would be useful, but I don't think that method belongs the in the `CRLP_FiscalYears` class.  Maybe it would make sense to have it in the `UTIL_String` class?

Reply:
Fair enough. Let's leave as is.